storage/aptosdb/src/state_store/mod.rs

```diff
@@ -137,6 +140,27 @@ impl DbReader for StateDb {
             .map(|(_, value)| value))
     }
 
+    /// Gets the latest state value and the its corresponding version when its of the given key up
+    /// to the given version.
+    fn get_state_value_with_version_by_version(
+        &self,
+        state_key: &StateKey,
+        version: Version,
+    ) -> Result<Option<(Version, StateValue)>> {
+        let mut read_opts = ReadOptions::default();
+        // We want `None` if the state_key changes in iteration.
+        read_opts.set_prefix_same_as_start(true);
+        let mut iter = self
+            .state_kv_db
+            .db_shard(state_key.get_shard_id())
+            .iter::<StateValueSchema>(read_opts)?;
+        iter.seek(&(state_key.clone(), version))?;
```

Comment:
do we want to seek from back and return the last one?

Reply:
It's @lightmark 's magic here: https://github.com/aptos-labs/aptos-core/blob/main/storage/aptosdb/src/schema/state_value/mod.rs#L43

Reply:
so the bigger version should be placed earlier than the smaller version.  wouldn't this result in a lot of reordering once new version of the same data got added. @lightmark ?

Reply:
It's anyway random update because the state_key is the first part of the key.